clients/fides-js/src/components/tcf/TcfPurposes.tsx

```diff
@@ -129,6 +118,8 @@ const TcfPurposes = ({
           onChange({ newEnabledIds, modelType: activeData.purposeModelType })
         }
         renderToggleChild={(purpose) => <PurposeDetails purpose={purpose} />}
+        // This key forces a rerender when legal basis changes, which allows paging to reset properly
```

Comment:
this is a neat pattern, more details here: https://react.dev/learn/you-might-not-need-an-effect#resetting-all-state-when-a-prop-changes

Reply:
oh very cool!